compositor_pipeline/src/queue.rs

```diff
@@ -57,55 +95,87 @@ pub struct Queue {
 impl Queue {
     pub fn new(output_framerate: Framerate) -> Self {
         Queue {
-            internal_queue: Mutex::new(InternalQueue::new()),
-            check_queue_channel: unbounded(),
+            video_queue: Mutex::new(VideoQueue::new()),
+            check_video_queue_channel: unbounded(),
             output_framerate,
             buffer_duration: DEFAULT_BUFFER_DURATION,
             clock_start: Instant::now(),
+            audio_queue: Mutex::new(AudioQueue::new()),
+            audio_chunk_duration: DEFAULT_AUDIO_CHUNK_DURATION,
         }
     }
 
-    pub fn add_input(&self, input_id: InputId) {
-        self.internal_queue.lock().unwrap().add_input(input_id);
+    pub fn add_input(&self, options: InputType) {
+        if options.video.is_some() {
```

Comment:
```suggestion
        if let Some(_) = options.video {
```

Reply:
I know that you want this to look consistent with the statement bellow, but:
https://rust-lang.github.io/rust-clippy/master/index.html#/redundant_pattern_matching

Reply:
Did you verify this? The pattern match bellow is also redundant, so what is the difference there?
If for some reason there is a differnce than maybe change audio check to is_some?

Reply:
Yep, I verified this.
It's interesting, since if I use
```rust
if let Some(_name) = sth {...}
```
Clippy doesn't have any issues, but 
```rust
if let Some(_) - sth {...}
```
throws a warning. 

I guess, that adding a `name` part might turn off the warning because someone wants to more thoroughly explain what is inside (I guess in some cases it might improve readability).

Anyway, using `Some(_audio_options)` when `_audio_options = ()` is a little bit weird, but I can use it if you want (IMO it doesn't make any difference here).

Reply:
doesn't matter which approach, but I would prefer the same one for both audio and video